.github/workflows/build_for_pypi.yml

```
@@ -1,46 +1,90 @@
+---
 name: Build and Optionally Publish to PyPi
 
 on:
-  workflow_call:
-    inputs:
-      publish:
-        type: boolean
-        default: false
-        description: "Publish to PyPi"
+  pull_request:
+    branches:
+      - main
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
 
 jobs:
-  build_for_pypi:
+  build_dist_for_pypi:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
```

Setting up Python isn't necessary, since the `cibuildwheel` will reset it first thing anyway: https://github.com/pypa/cibuildwheel/blob/3f123af/action.yml#L28-L32

One thing I'd recommend would be feeding the sdist into cibuildwheel. So best download it here:

```suggestion
      - name: Download the sdist
        uses: actions/download-artifact@v4
        with:
          name: cibw-sdist
```